test/AngularSpec.js

```diff
@@ -78,6 +78,42 @@ describe('angular', function() {
       expect(copy(objWithRegExp.re) === objWithRegExp.re).toBeFalsy();
     });
 
+    it("should copy a TypedArray with no destination", function() {
+      if (typeof Float32Array !== 'undefined') {
+        var arraysToCheck = [
+          Uint8Array, Uint16Array, Uint32Array, Int8Array, Int16Array,
+          Int32Array, Float32Array, Float64Array
+        ];
+
+        for (var i = 0; i < arraysToCheck.length; i++) {
+          var TypeOfArray = arraysToCheck[i];
+          // only run this test on browsers that support typed arrays
+          var src = new TypeOfArray(2);
+          src[1] = 1;
+          var dst = copy(src);
+          expect(copy(src) instanceof TypeOfArray).toBeTruthy();
+          expect(dst).toEqual(src);
+          expect(dst).not.toBe(src);
+        }
+      }
+    });
+
+    it("should throw an exception if a TypedArray is the destination", function() {
+      if (typeof Float32Array !== 'undefined') {
+        var arraysToCheck = [
+          Uint8Array, Uint16Array, Uint32Array, Int8Array, Int16Array,
```

Comment:
@caitp I can split these, it's not big deal, don't wanna commit code that would have to be cleaned up later :)